internal/expr/decode.go

```diff
@@ -0,0 +1,63 @@
+package expr
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+// markerIndex returns the int X from the string "_sqlair_X".
+func markerIndex(s string) (int, bool) {
+	if strings.HasPrefix(s, markerPrefix) {
+		n, err := strconv.Atoi(s[len(markerPrefix):])
+		if err == nil {
+			return n, true
+		}
+	}
+	return 0, false
+}
+
+// ScanArgs returns list of pointers to the struct fields that are listed in qe.outputs.
+// All the structs mentioned in the query must be in outputVals.
+// All outputVals must be of kind reflect.Struct.
+func (qe *QueryExpr) ScanArgs(columns []string, outputVals []reflect.Value) ([]any, error) {
```

Comment:
In the previous review, ScanArgs actually takes outputArgs in by itself. The reason for this was that all the reflection manipulation is being done inside the expr package currently. It seems awkward to have that one detail being done outside, when even the values themselves are not used there but here.

Reply:
True, I've moved it inside `ScanArgs`